src/database-controller/src/write-merger/app.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+const cors = require('cors');
+const morgan = require('morgan');
+const express = require('express');
+const compress = require('compression');
+const bodyParser = require('body-parser');
+const logger = require('@dbc/core/logger');
+const status = require('statuses');
+const handler = require('@dbc/write-merger/handler');
+const config = require('@dbc/write-merger/config');
+
+const app = express();
+
+app.use(cors());
+app.use(compress());
+app.use(bodyParser.urlencoded({ extended: true }));
+app.use(bodyParser.json({ limit: config.bodyLimit }));
+app.use(bodyParser.text({ type: 'text/*' }));
+app.use(morgan('dev', { stream: logger.stream }));
+
+const router = new express.Router();
+
+router.route('/ping').get(handler.ping);
+router.route('/frameworkRequest').put(handler.receiveFrameworkRequest);
+router.route('/watchEvents/:eventType').post(handler.receiveWatchEvents);
+
+app.use('/api/v1', router);
+
+// error handling
+app.use((err, req, res, next) => {
+  logger.warn(err.stack);
+  const statusCode = err.statusCode || 500;
+  res.status(statusCode).json({
+    code: status(statusCode),
+    message: err.message,
+  });
```

Comment:
return the whole err in response ?

Reply:
This error will be shown in the webportal. Because normal users don't care about the error stack, I don't return the whole error object here. For developer, he/she can see the error stack from log.

Reply:
ok, but not only restserver call it, watch also call

Reply:
Yes, but the error is recorded in write merger at least.